src/installer/corehost/corehost.cpp

```diff
@@ -84,6 +83,14 @@ bool is_exe_enabled_for_execution(pal::string_t* app_dll)
 #define CURHOST_EXE
 #endif
 
+void need_newer_framework_error()
+{
+    pal::string_t url = get_download_url();
+    trace::error(_X("  _ To run this application, you need to install a newer version of .NET Core."));
+    trace::error(_X(""));
+    trace::error(_X("  - %s"), url.c_str());
```

Comment:
Can you please append the apphost version to the URL? The problem is that without it there's no version information in the URL and the web server will have to way of telling at least roughly which version to offer. Once we ship 6.0 that would probably become the default and 5.0 apps would not work with it (default roll forward is Minor only). It would require two installations (6.0 first - which would provide better error reporting as we would load hostfxr and then 5.0 to actually run the app).

See this change which improved this for the other cases: https://github.com/dotnet/runtime/pull/33602